controllers/harvestercluster_controller_test.go

```diff
@@ -51,3 +66,187 @@ users:
+`
+	fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "harvester-csi-driver-addon",
+				Namespace: "test-hv",
+			},
+			Data: map[string]string{"harvester-cloud-provider-deploy.yaml": manifest},
+		}).Build()
+	var r *HarvesterClusterReconciler
+	var scope *ClusterScope
+	var log logr.Logger = log.FromContext(context.TODO())
+	BeforeEach(func() {
+		hvConfig, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+		Expect(err).To(BeNil())
+		hvClient, err := hvclient.NewForConfig(hvConfig)
+		Expect(err).To(BeNil())
+
+		r = &HarvesterClusterReconciler{
+			Client: fakeClient,
+			Scheme: scheme,
+		}
+
+		scope = &ClusterScope{
+			Ctx:    context.TODO(),
+			Logger: log,
+			Cluster: &clusterv1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-2",
+					Namespace: "test-hv",
+				},
+				Spec: clusterv1.ClusterSpec{},
+			},
+			HarvesterCluster: &infrav1.HarvesterCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-2-hv",
+					Namespace: "test-hv",
+				},
+				Spec: infrav1.HarvesterClusterSpec{
+					TargetNamespace: "default",
+					Server:          "https://192.168.1.109:6443",
+					UpdateCloudProviderConfig: infrav1.UpdateCloudProviderConfig{
+						ManifestsConfigMapNamespace:      "test-hv",
+						ManifestsConfigMapName:           "harvester-csi-driver-addon",
```

Comment:
Looks like the cloud provider manifest is currently in the `harvester-csi-driver-addon` config map, while the csi manifest in the `cloud-controller-manager-addon` config map. Do we want an issue to fix it?